internal/driver/driver.go

```diff
@@ -121,12 +133,51 @@ func (d *Driver) Initialize(sdk interfaces.DeviceServiceSDK) error {
 
 	// Make sure the paths of existing devices are up-to-date.
 	go d.RefreshExistingDevicePaths()
-
+	go d.StartRtspCredentialServer()
 	return nil
 }
+func (d *Driver) StartRtspCredentialServer() {
+	d.lc.Infof("starting rtsp server")
+	rtspAuthServer := http.NewServeMux()
+	rtspAuthServer.HandleFunc("/rtspauth", d.RTSPCredentialsHandler)
+	http.ListenAndServe(d.rtspAuthenticationServer, rtspAuthServer)
```

Comment:
I think we probably need to add code to Stop the server on `Stop()`

Reply:
should i do something like this https://github.com/rudderlabs/graceful-shutdown-examples/blob/main/httpserver/main.go

Reply:
@brianointel A little similar, but I think it can be done cleaner by adding a waitgroup and a pointer to the http server in the driver struct. In initialize(), before calling `go StartRTSP...` function, do rtspAuthWaitGroup.Add(1), and as the first line of this StartRTSP... func, do `defer rtspAuthWaitGroup.Done()`. Then in the driver.Stop() function, shutdown the server, and then call `rtspAuthWaitGroup.Wait()`.